crates/validation_common/src/lib.rs

```diff
@@ -0,0 +1,114 @@
+// SPDX-FileCopyrightText: 2022 Profian Inc. <opensource@profian.com>
+// SPDX-License-Identifier: AGPL-3.0-only
+
+mod digest;
+
+pub use digest::Digest;
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[serde(untagged)]
+pub enum Measurements {
+    /// Only Enarx signer information is available. This is the key which
+    /// has signed the Enarx binary, directly or indirectly.
+    Signer { enarx_signer: Vec<Digest> },
+
+    /// This is the hash of the Enclave environment after the Enarx binary is loaded
+    /// but before any workload is loaded, so this is a hash of the Enarx binary
+    /// in memory.
+    Hash { enarx_hash: Vec<Digest> },
+
+    /// Both the Enarx hash and hash of a signing key are present.
+    Both {
+        enarx_hash: Vec<Digest>,
+        enarx_signer: Vec<Digest>,
+    },
```

Comment:
So, the only useful combination of signer and hash would be a signer whitelist and hash blacklist IMHO